### PR TITLE
ci: test on Node.js 22

### PR DESCRIPTION
**Pull request by zkochan**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ jobs:
         node:
           - '18.12'
           - '20'
-          - '21'
```
Comment (weyert) [inline .github/workflows/ci.yml:20]: We don't want to test against v21?

**Review by weyert — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -43,8 +43,8 @@ jobs:
       with:
         node-version: ${{ matrix.node }}
         cache: 'pnpm'
-    - name: Install npm@7
-      run: npm add --global npm@7
+    - name: Install npm@10
```
Comment (weyert) [inline .github/workflows/ci.yml]: npm v10 appears to require a minimum version of Node v20. 

Do we want to do a matrix of `node-version` and `npc-version` to accommodate for this?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (zkochan) [reply .github/workflows/ci.yml:20]: v21 is not an lts version. When 22 is out, v21 is basically deprecated.

Comment (zkochan) [reply .github/workflows/ci.yml:20]: as you can see, we don't test v19 as well.

**Pull request merged**